service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -76,6 +85,14 @@ public MethodsApiController(
 
   @Override
   public ResponseEntity<PostMethodResponse> postMethod(PostMethodRequest postMethodRequest) {
+    // check if current user has write permissions on the workspace
+    if (!samService.hasWritePermission()) {
+      String errorMsg =
+          getSamForbiddenExceptionMsg(SamService.WRITE_ACTION, SamService.RESOURCE_TYPE_WORKSPACE);
+      logger.info(errorMsg);
```

Comment:
Do we need to log messages as well as throw the exception? When I run this locally, I see this logged message, but I also see the messages logged by the `GlobalExceptionHandler`, and the response contains the proper message as well. I think we may be able to get rid of the `logger.info` and let GlobalExceptionHandler deal with the messageExtraction. Then we could move the ForbiddenExceptionMsg to the ForbiddenException itself, and pass in the Action and ResourceType to the constructor. I think this would help reduce duplication across controller/endpoints as well.

Reply:
Yeah I added this log message with the thought that when we fetch user IDs we can log the user ID here, but not send the user ID back to the user via the exception message.

Reply:
Since each server log, including exceptions would be logged with user ID in a later PR, the log messages added were redundant. Hence I have removed it.